.github/workflows/integration_test.yaml

```diff
@@ -267,7 +267,8 @@ jobs:
     runs-on: ubuntu-latest
     needs:
       - integration-test
-    if: always()
+    if: always() && !cancelled()
+    timeout-minutes: 5
```

Comment:
Am I missing what this timeout means? Because 5 min doesn't seem to suffice for most cases

Reply:
Added the timeout just for the check jobs which just spin up and run a few bash commands. It should be safe, if they are stuck for some reason this will cancel them much sooner